eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -398,7 +398,7 @@ class Router(nodeParams: NodeParams, watcher: ActorRef, initialized: Option[Prom
 
     case Event(FinalizeRoute(partialHops), d) =>
       // split into sublists [(a,b),(b,c), ...] then get the edges between each of those pairs, then select the largest edge between them
-      val edges = partialHops.sliding(2).map { case List(v1, v2) => d.graph.getEdgesBetween(v1, v2).maxBy(_.update.htlcMaximumMsat) }
+      val edges = partialHops.sliding(2).map { case List(v1, v2) => d.graph.getEdgesBetween(v1, v2).maxBy(_.update.htlcMaximumMsat.getOrElse(0 msat)) }
```

Comment:
I think this is unnecessary since `maxBy` can work with options too, for example this works in the REPL:
```
scala> case class Foo(elem: Option[Int])
defined class Foo

scala> List(Foo(Some(1)), Foo(Some(2)), Foo(None))
res2: List[Foo] = List(Foo(Some(1)), Foo(Some(2)), Foo(None))

scala> res2.maxBy(_.elem)
res3: Foo = Foo(Some(2))

scala> res2.minBy(_.elem)
res4: Foo = Foo(None)
```
However the `maxBy` operation can't be performed on empty lists and if this happens this like will throw an exception (this happened in the wild already).

Reply:
It doesn't compile if I remove it because of evil implicits saying it's ambiguous to choose the right `Ordering`...